src/main/java/org/jetbrains/id/names/suggesting/impl/IdNamesNGramModelRunner.java

```diff
@@ -105,15 +136,43 @@ private int predictUsageName(@NotNull List<Prediction> predictionList,
                         toProb(e.getValue()),
                         usagePriority))
                 .sorted((pred1, pred2) -> -Double.compare(pred1.getProbability(), pred2.getProbability()))
-                .limit(IdNamesSuggestingService.PREDICTION_CUTOFF)
+                // Anyway predictions will be filtered later.
+                .limit(IdNamesSuggestingService.PREDICTION_CUTOFF + 10)
```

Comment:
Why 10? It might be better to return them all and decide how to trim them in a single place.

Reply:
Memory efficiency?) There might be too much predictions for an identifier, but may be I've overthought this.